### PR TITLE
Update dataloader.py

### PR DESCRIPTION
**Pull request by MatthewBM**
I have dataloaders that stream tensors from the internet.

Sometimes the daemons they communicate with are unresponsive towards a specific worker, and the worker can timeout.

I would like to handle these errors by resetting the dataloader to allow models to continue, however the class of error is not specifically defined in this version of Pytorch, and it instead uses a generic Runtime Error, 

This merge would fix that with the specific TimeoutError exception

### CODE REVIEW

**Review by MatthewBM — COMMENTED** [review]
Looks Great!

**Comment by ejguan** [comment]
Hi @MatthewBM ,

Thanks for being interested in contributing PyTorch. 
I am sorry that I have to close this PR based on: https://github.com/pytorch/pytorch/wiki/Developer-FAQ#my-pr-was-rejected-for-being-too-small--what-gives

You can find issues with label `good first issue` to start contributing PyTorch: https://github.com/pytorch/pytorch/labels/good%20first%20issue
